src/main/java/kitchenpos/domain/MenuProduct.java

```diff
@@ -1,48 +1,57 @@
 package kitchenpos.domain;
 
+import java.math.BigDecimal;
+
 public class MenuProduct {
-    private Long seq;
-    private Long menuId;
-    private Long productId;
-    private long quantity;
+    private final Long seq;
+    private final Long menuId;
+    private final Long productId;
+    private final long quantity;
+
+    private final BigDecimal price;
 
-    public MenuProduct(Long productId, long quantity) {
+    private MenuProduct(Long seq, Long menuId, Long productId, long quantity, BigDecimal price) {
+        this.seq = seq;
+        this.menuId = menuId;
         this.productId = productId;
         this.quantity = quantity;
+        this.price = price;
     }
 
-    public MenuProduct() {
+    public MenuProduct(Long seq, Long menuId, Long productId, long quantity) {
+        this(seq, menuId, productId, quantity, null);
     }
 
-    public Long getSeq() {
-        return seq;
+    public MenuProduct(Long menuId, Long productId, long quantity) {
+        this(null, menuId, productId, quantity, null);
     }
 
-    public void setSeq(final Long seq) {
-        this.seq = seq;
+    public MenuProduct(Long productId, BigDecimal price, long quantity) {
+        this(null, null, productId, quantity, price);
     }
 
-    public Long getMenuId() {
-        return menuId;
+    public Long getSeq() {
+        return seq;
     }
 
-    public void setMenuId(final Long menuId) {
-        this.menuId = menuId;
+    public Long getMenuId() {
+        return menuId;
     }
 
     public Long getProductId() {
         return productId;
     }
 
-    public void setProductId(final Long productId) {
-        this.productId = productId;
-    }
-
     public long getQuantity() {
         return quantity;
     }
 
-    public void setQuantity(final long quantity) {
-        this.quantity = quantity;
+    public BigDecimal getPrice() {
+        return price;
+    }
+
+    public BigDecimal getAmount() {
+        BigDecimal multiply = price.multiply(BigDecimal.valueOf(quantity));
+        return multiply;
```

Comment:
inline으로 바로 반환해도 될 것 같아요!
```suggestion
    public BigDecimal getAmount() {
        return price.multiply(BigDecimal.valueOf(quantity));
```